### PR TITLE
-replaced expand icons with  ExpandButtonTemplate.

### PR DESCRIPTION
**Pull request by SalehSabrah**
-added new property to show expand icon for leave node to support on-demand population of nodes using the expand event.
-added DoubleClick event
-changed SelectedBackgroundColor & Opacity to bindable properties.

### CODE REVIEW

**Review by MelbourneDeveloper — COMMENTED** [review]
Hey. Sorry I haven't gotten around to looking at this. It looks like you've really ticked off a few things that I wanted to do. It may even help to get it to the point where it can be submitted to the XF repo. Anyway, feel free to keep working in your fork, and I will eventually merge whatever I can in to this.

**Comment by SalehSabrah** [comment]
Sounds good. Thanks.

File: Adapt.PresentationSamples.Standard/MainPage.xaml.cs [hunk 1]
```diff
@@ -96,9 +96,50 @@ private static TreeViewNode CreateTreeViewNode(object bindingContext, Label labe
                     Orientation = StackOrientation.Horizontal
                 }
             };
+            node.SelectedBackgroundColor;
+
+            //set DataTemplate for expand button content
+            node.ExpandButtonTemplate = new DataTemplate(() => new ExpandButtonContent { BindingContext = node});
```
Comment (MelbourneDeveloper) [inline Adapt.PresentationSamples.Standard/MainPage.xaml.cs:102]: Smart move putting this and extricating it from the code! This is what I had planned. Nice.

**Comment by MelbourneDeveloper** [comment]
I think this is all good. Just a couple of little things to fix up in the comments.

**Comment by SalehSabrah** [comment]
Thanks for looking into this. Please let me know if I need to do something on my end as I am new to GitHub.

**Comment by MelbourneDeveloper** [comment]
No problem, If you just fix the couple of small issues I marked, I will accept the pull request. I guess the question is what to do with this afterwards? Get the TreeView in to a NuGet package?

**Comment by SalehSabrah** [comment]
Thanks for committing the changes.  I agree it would be nice to have it a NuGet package.  I can confirm that it is working as expected in IOS. At some point I will have it tested in Android as well.

**Comment by MelbourneDeveloper** [comment]
This is awesome. We are gearing up to do a bunch of iOS work so this will be really helpful for us. I'll chat in our team and decide how best to release this.

**Pull request merged**